app/src/main/java/com/keylesspalace/tusky/components/systemnotifications/NotificationFetcher.kt

```diff
@@ -72,69 +72,40 @@ class NotificationFetcher @Inject constructor(
-                        )
-                            .forEach { notificationManager.cancel(it.tag, it.id) }
-
-                        // Still got notifications to remove? Trim the list of new notifications,
-                        // starting with the oldest.
-                        while (notifications.size > MAX_NOTIFICATIONS) {
-                            notifications.removeAt(0)
-                        }
-                    }
+                    // NOTE: There is no need to manage notification numbers: The system will
+                    //   by itself display the newest up to its own internal (and unknown) limit.
 
-                    val notificationsByType = notifications.groupBy { it.type }
+                    val notificationsByType: Map<Notification.Type, List<Notification>> = notifications.groupBy { it.type }
 
-                    // Make and send the new notifications
-                    // TODO: Use the batch notification API available in NotificationManagerCompat
-                    // 1.11 and up (https://developer.android.com/jetpack/androidx/releases/core#1.11.0-alpha01)
-                    // when it is released.
-
-                    notificationsByType.forEach { notificationsGroup ->
+                    var typeIndex = 0
+                    notificationsByType.forEach { notificationsGroup: Map.Entry<Notification.Type, List<Notification>> ->
                         notificationsGroup.value.forEach { notification ->
                             val androidNotification = NotificationHelper.make(
                                 context,
                                 notificationManager,
                                 notification,
-                                account,
-                                notificationsGroup.value.size == 1
+                                account
                             )
                             notificationManager.notify(
                                 notification.id,
                                 account.id.toInt(),
                                 androidNotification
                             )
+                        }
+
+                        NotificationHelper.updateSummaryNotification(
+                            context,
+                            notificationManager,
+                            account,
+                            notificationsGroup.key
+                        )
 
-                            // Android will rate limit / drop notifications if they're posted too
-                            // quickly. There is no indication to the user that this happened.
-                            // See https://github.com/tuskyapp/Tusky/pull/3626#discussion_r1192963664
+                        // Have a small pause before the next channel = obvious (audible, ...) notification
+                        typeIndex++
+                        if (typeIndex < notificationsByType.size)
```

Comment:
We could get rid of this delay as well: We normally only have 1-3 summary notifications and if they appear without delay from each other the would probably only ring once.

Reply:
Good idea, let's try it

Reply:
done